File: androidApp/src/main/kotlin/com/riox432/civitdeck/ui/components/ModelStatsRow.kt
package com.riox432.civitdeck.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Download
import androidx.compose.material.icons.outlined.FavoriteBorder
import androidx.compose.material.icons.outlined.Star
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.dp
import com.riox432.civitdeck.ui.theme.IconSize
import com.riox432.civitdeck.ui.theme.Spacing
import com.riox432.civitdeck.util.FormatUtils

/**
 * Displays model statistics (downloads, favorites, rating, optional comments).
 *
 * When [commentCount] is null, renders a compact icon+value row (3 items).
 * When [commentCount] is provided, renders an expanded value+label column row (4 items).
 */
@Composable
fun ModelStatsRow(
    downloadCount: Int,
    favoriteCount: Int,
    rating: Double?,
    commentCount: Int? = null,
    modifier: Modifier = Modifier,
) {
    if (commentCount != null) {
        // Expanded (detail) style: value above label, spaced evenly
        Row(
            modifier = modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceEvenly,
        ) {
            StatColumn(
                value = FormatUtils.formatCount(downloadCount),
                label = "Downloads",
            )
            StatColumn(
                value = FormatUtils.formatCount(favoriteCount),
                label = "Favorites",
            )
            StatColumn(
                value = if (rating != null) FormatUtils.formatRating(rating) else "-",
                label = "Rating",
            )
            StatColumn(
                value = FormatUtils.formatCount(commentCount),
                label = "Comments",
            )
        }
    } else {
        // Compact style: icon + value inline
        Row(
            modifier = modifier,
            horizontalArrangement = Arrangement.spacedBy(Spacing.sm),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            StatItem(
                label = FormatUtils.formatCount(downloadCount),
                icon = Icons.Outlined.Download,
            )
            StatItem(
                label = FormatUtils.formatCount(favoriteCount),
                icon = Icons.Outlined.FavoriteBorder,
            )
            StatItem(
                label = if (rating != null) FormatUtils.formatRating(rating) else "-",
                icon = Icons.Outlined.Star,
            )
        }
    }
}

@Composable
private fun StatItem(
    label: String,
    icon: ImageVector,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier,
        horizontalArrangement = Arrangement.spacedBy(2.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            modifier = Modifier.size(IconSize.statIcon),
            tint = MaterialTheme.colorScheme.onSurfaceVariant,
        )
        Text(
            text = label,
            style = MaterialTheme.typography.labelSmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
    }
}

@Composable
private fun StatColumn(value: String, label: String) {
    Column(horizontalAlignment = Alignment.CenterHorizontally) {
        Text(
            text = value,
            style = MaterialTheme.typography.titleMedium,
        )
        Text(
            text = label,
            style = MaterialTheme.typography.labelSmall,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
    }
}

File: iosApp/iosApp/DesignSystem/ModelStatsRow.swift
import SwiftUI

/// Displays model statistics (downloads, favorites, rating, optional comments).
///
/// When `commentCount` is nil, renders a compact icon+value row (3 items).
/// When `commentCount` is provided, renders an expanded value+label column row (4 items).
struct ModelStatsRow: View {
    let downloadCount: Int
    let favoriteCount: Int
    let rating: Double?
    var commentCount: Int? = nil

    var body: some View {
        if let commentCount {
            expandedRow(commentCount: commentCount)
        } else {
            compactRow
        }
    }

    // MARK: - Compact (icon + value inline)

    private var compactRow: some View {
        HStack(spacing: Spacing.sm) {
            statItem(icon: "arrow.down.circle",
                     value: FormatUtils.shared.formatCount(count: downloadCount))
            statItem(icon: "heart",
                     value: FormatUtils.shared.formatCount(count: favoriteCount))
            statItem(icon: "star",
                     value: ratingText)
        }
    }

    private func statItem(icon: String, value: String) -> some View {
        HStack(spacing: 2) {
            Image(systemName: icon)
                .font(.system(size: IconSize.statIcon))
            Text(value)
                .font(.civitLabelSmall)
        }
        .foregroundColor(.civitOnSurfaceVariant)
    }

    // MARK: - Expanded (value above label, spaced evenly)

    private func expandedRow(commentCount: Int) -> some View {
        HStack {
            statColumn(value: FormatUtils.shared.formatCount(count: downloadCount),
                       label: "Downloads")
            Spacer()
            statColumn(value: FormatUtils.shared.formatCount(count: favoriteCount),
                       label: "Favorites")
            Spacer()
            statColumn(value: ratingText, label: "Rating")
            Spacer()
            statColumn(value: FormatUtils.shared.formatCount(count: commentCount),
                       label: "Comments")
        }
    }

    private func statColumn(value: String, label: String) -> some View {
        VStack(spacing: 2) {
            Text(value)
                .font(.civitTitleMedium)
            Text(label)
                .font(.civitLabelSmall)
                .foregroundColor(.civitOnSurfaceVariant)
        }
    }

    // MARK: - Helpers

    private var ratingText: String {
        guard let rating else { return "-" }
        return FormatUtils.shared.formatRating(rating: rating)
    }
}
